app/validation/validators.py

```diff
@@ -295,23 +296,29 @@ def _is_first_relative_delta_largest(relativedelta1, relativedelta2):
     def _build_range_length_error(self, period_object):
         error_message = ''
         if 'years' in period_object:
-            error_message = self.return_error_component(error_message, period_object['years'], ' year')
+            error_message = self.return_error_component(error_message, period_object['years'], 'year')
         if 'months' in period_object:
-            error_message = self.return_error_component(error_message, period_object['months'], ' month')
+            error_message = self.return_error_component(error_message, period_object['months'], 'month')
         if 'days' in period_object:
-            error_message = self.return_error_component(error_message, period_object['days'], ' day')
+            error_message = self.return_error_component(error_message, period_object['days'], 'day')
 
         return error_message
 
     @staticmethod
     def return_error_component(error_message, number, unit):
+        unit_plurals = {
+            'year': [_('years'), _('year')],
+            'month': [_('months'), _('month')],
+            'day': [_('days'), _('day')]
+        }
+        plural = unit_plurals[unit][(1 if number == 1 else 0)]
```

Comment:
Welsh has 4 plural forms (`Plural-Forms: nplurals=4; plural=(n==1) ? 0 : (n==2) ? 1 : (n != 8 && n != 11) ? 2 : 3;\n`)

E.g. (from google translate)
0 years: 0 blynedd
1 year: 1 flynedd
2 years: 2 flynedd
3 (and 4 and 6) years: 3 blynedd
5 (and all other years): 5 mlynedd

gettext supports plural forms natively with [ngettext](https://pythonhosted.org/Flask-Babel/#flask.ext.babel.ngettext).

I'm a little concerned with all the string concatenation that we're going to either get untranslated parts of validation error messages or we'll end up preventing a translator from constructing the text correctly.

Here's some english -> welsh examples:

"Enter a reporting period greater than or equal to 0 years." -> "Rhowch gyfnod adrodd yn fwy na neu'n hafal i 0 blynedd."
"Enter a reporting period greater than or equal to 1 years." -> "Rhowch gyfnod adrodd yn fwy nag 1 flwyddyn neu fwy cyfartal."
"Enter a reporting period greater than or equal to 2 years." -> "Rhowch gyfnod adrodd yn fwy na 2 flynedd neu gyfartal."

Note that there's extra differences beyond the word for "year" that get altered depending on the count.

I think the only way we can reasonably do this is to define all of the error messages in their complete forms and accepting only a numeric count as an argument rather than a translated unit.

"Enter a reporting period greater than or equal to %(min)s years."
"Enter a reporting period greater than or equal to %(min)s months."
"Enter a reporting period greater than or equal to %(min)s days."

Reply:
@nealedj We would actually need an error message for each combination or years, months, days which is huge.

Reply:
Yeah let's break that out into a separate job in that case.

I think it's worth looking at the pluralization of year(s), month(s) and day(s) though. We can make that a little more translatable with `ngettext`